Assets/Scripts/EnemyController.cs

```diff
@@ -0,0 +1,127 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.AI;
+using System.Collections;
+using Photon.Pun;
+
+public class EnemyController : MonoBehaviourPunCallbacks, IPunObservable
+{
+    public GameObject[] players;
+    public Transform player;
+    private float distanceToPlayer;
+    public int minDist = 2;
+    public int speed = 2;
+    private bool isAttackReady = true;
+    private float attackAnimationDelay = .5f;
+
+    public float health = 50f;
+
+    private NavMeshAgent agent;
+    //public Animator animator;
+    private int refreshTargetTimer = 0;
+    public int refreshTargetTimerLimit = 50;
+
+    void Start()
+    {
+        //animator = GetComponentInChildren<Animator>();    //Uncomment when adding animator
+        agent = GetComponent<NavMeshAgent>();
+        //animator.Play("AngryFlight");     //Walking animation
+        agent.stoppingDistance = 2;
+        players = findPlayers();
+    }
+
+    void Update()
+    {
+        if (health < 0)
+        {
+            PhotonNetwork.Destroy(gameObject);
+        }
+        FindNavTarget();
+        distanceToPlayer = Vector3.Distance(player.position, transform.position);
+        SetSpeed();
+        Attack();
+    }
+
+    void SetSpeed()
+    {
+        agent.speed = speed;
+    }
+
+    void FindNavTarget()
+    {
+        refreshTargetTimer -= 1;
+
+        //we set destination for target to run less than every frame, cause it's computationally heavy over longer distances
+        if (refreshTargetTimer <= 0)
+        {
```

Comment:
maybe using `InvokeRepeatedly` or something like that would be better practice for that? We could repeatedly check for the closest player every half a second or something like that

Reply:
Ooh I didn't know about this one, I will definitely use it from now on ❤️

Reply:
Realised that we can't use it because of networking :D

Reply:
Oh crap 😓